From 39eebb652db570cf51994f404da3dd91c8a9f4cc Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Fri, 21 Jul 2017 23:25:35 +0800
Subject: [PATCH] Fix compatibility with tornado 4.3+

`HTTPHeaders` is not JSON serializable, so make it a dict first.
---
 dummyserver/handlers.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dummyserver/handlers.py b/dummyserver/handlers.py
index a943b836b..b91fe7215 100644
--- a/dummyserver/handlers.py
+++ b/dummyserver/handlers.py
@@ -225,7 +225,7 @@ def encodingrequest(self, request):
         return Response(data, headers=headers)
 
     def headers(self, request):
-        return Response(json.dumps(request.headers))
+        return Response(json.dumps(dict(request.headers)))
 
     def successful_retry(self, request):
         """ Handler which will return an error and then success
